risc0/zkvm/src/exec/env.rs

```diff
@@ -146,8 +143,11 @@ impl<'a> ExecutorEnvBuilder<'a> {
     }
 
     /// Set a segment limit, specified in powers of 2 cycles.
+    ///
+    /// If the given value exceeds [risc0_zkp::MAX_CYCLES_PO2], then the value
+    /// [risc0_zkp::MAX_CYCLES_PO2] is used instead.
     pub fn segment_limit_po2(&mut self, limit: usize) -> &mut Self {
-        self.inner.segment_limit_po2 = limit;
+        self.inner.segment_limit_po2 = usize::min(limit, risc0_zkp::MAX_CYCLES_PO2);
```

Comment:
This might be confusing to users. Maybe we should make the `build` function be failable and so it can do integrity checks on the env and return if its valid or not.

Reply:
Good question. I agree that the behavior as-written might be confusing (I included a note in the docs, but it's still likely to confuse people who haven't read that note).

We could make it do that `build()` returns `Result<>`. We could also make it so that `segment_limit_po2()` returns `Result<>`, depending on where we'd want to surface the error.

I don't know how to judge between these options :/ What do you think would be best?

Reply:
I think we break the builder pattern if we mage `segment_limit_po2()` return a Result, I'm actually not sure how that would work.

So I think if we return a `Result` from `build`, then we have the ability to do verification checks for this, and beyond.

Reply:
`Result<>` from `build()` sounds reasonable to me. I'll update the PR